pkg/process/procutil/process_unsupported.go

```diff
@@ -26,15 +27,28 @@ func WithPermission(enabled bool) Option {
 	}
 }
 
+// WithCollectStats configures whether the probe should collect stats.
+// If collectStats is false, the probe will only collect process metadata
+func WithCollectStats(enabled bool) Option {
```

Comment:
It's a bit counter intuitive, a better option would be to just pass the boolean to the `ProcessesByPID` method

Reply:
I think having options on `Probe` object itself seems more like adding configurations than keep adding parameters to a function, plus some other configurations are already having similar pattern.

Reply:
The pattern of using options is fine, it's just the application of it to changing the behavior of a single method for `Probe` is what I found to be unintuitive, consider this - `Probe` has these methods:
```
StatsForPIDs(pids []int32, now time.Time) (map[int32]*Stats, error)
ProcessesByPID(now time.Time) (map[int32]*Process, error)
StatsWithPermByPID(pids []int32) (map[int32]*StatsWithPerm, error)
```

We are changing the behavior of `ProcessesByPID` in this case as opposed to introducing an option universally applicable to all three.

What makes more sense to me is changing this method only:
```
ProcessesByPID(now time.Time, collectStats bool) (map[int32]*Process, error)
```

Reply:
I definitely agree with @xornivore here. It definitely wouldn't make sense in the context of a method like `StatsForPIDs` where the entire goal is to collect stats. If it is enabled should `StatsForPIDs` work as expected, or do nothing? It is unclear.

I'll go ahead and make this change 😄

Reply:
Should be fixed now!